src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -564,9 +579,10 @@ def build(
             ), "No schema query is available in child SnowflakePlan"
             new_schema_query = schema_query or sql_generator(child.schema_query)
 
+        config_context = self._config_context or ConfigContext(self.session)
```

Comment:
each build function seems can take a new snapshot, but multiple build can be called during the whole resolve, and they are suppose to get the same snapshot, it is safer and cleaner to simply extend each interface to take the config context.

Reply:
I intend to remove `config_context` from plan_builder once old CTE implementation is gone. That is the only reason why plan builder needs this info. I can own this task. Without that, the refactor will be too big and it would be prone to more errors. I want to avoid that.